packages/browser-wallet/src/popup/pages/AddWeb3IdCredential/AddWeb3IdCredential.tsx

```diff
@@ -115,23 +116,31 @@ export default function AddWeb3IdCredential({ onAllow, onReject }: Props) {
 
     useEffect(() => {
         if (schema) {
-            // Ensure that all attributes required by the schema are in the attributes. If not, then
-            // the credential should not be allowed to be added.
-            const missingRequiredAttributeKeys = [];
-            const requiredAttributes = schema.properties.credentialSubject.properties.attributes.required;
-            for (const requiredAttribute of requiredAttributes) {
-                if (!Object.keys(credential.credentialSubject.attributes).includes(requiredAttribute)) {
-                    missingRequiredAttributeKeys.push(requiredAttribute);
+            // Use the schema to validate the credential.
+            const validator = new Validator();
+            try {
+                const validationResult = validator.validate(
+                    { credentialSubject: credential.credentialSubject },
+                    schema as unknown as Schema
```

Comment:
This does allow me to do timestamp attributes as dates. Did we want allow that here or only in statements?

Reply:
I am not sure.

Reply:
Ah, I understand now. It does not allow it now. Also if I add the object type, then it is not displayed correctly (shows as [object Object].

Reply:
Going to as the core team about it. We will leave this out until we know.

Reply:
@shjortConcordium we decide how we want it. So I would be interested in your opinion. Restricting to the schema when adding would also fix the serialization issue we had (and already fixed).